### PR TITLE
Replaced Delay futures with single DelayQueue for quorum deadline

### PR DESCRIPTION
**Pull request by echupriyanov**
Using Delay() future pollutes runtime with many unnecessary futures. Using DelayQueue is more straightforward and allows clearing delay, if all keys were received before deadline

### CODE REVIEW

File: randhound/src/lib.rs [hunk 1]
```diff
@@ -304,12 +306,19 @@ impl Future for RandHoundService {
                     RandHoundEvent::Epoch(msg) => self.on_epoch(msg),
                     RandHoundEvent::Heartbeat(msg) => self.on_heartbeat(msg),
                     RandHoundEvent::Timer(_i) => self.on_timer(),
-                    RandHoundEvent::CheckInitQuorum => self.on_check_init_quorum(),
                     RandHoundEvent::NewRound => self.on_new_round(),
                     RandHoundEvent::Subscribe(tx) => self.state.subscribe(tx),
                 },
                 Ok(Async::Ready(None)) => unreachable!(), // never should happen
-                Ok(Async::NotReady) => return Ok(Async::NotReady),
+                Ok(Async::NotReady) => match self.state.poll_deadline() {
```
Comment (rtsisyk) [inline randhound/src/lib.rs]: Why deadlines are polled only when main loop doesn't have events? All timers and deadlines are usually more important when network events.

File: randhound/src/randhound.rs [hunk 2]
```diff
@@ -306,13 +304,22 @@ pub(crate) fn init_state(
...
         msg_queue: VecDeque::new(),
-        runtime,
         service,
         consumers: vec![],
+        deadline: DelayQueue::new(),
     }
 }
 
 impl GlobalState {
+    pub fn poll_deadline(&mut self) -> Poll<Option<Hash>, Error> {
```
Comment (rtsisyk) [inline randhound/src/randhound.rs]: I don't understand why this part of RandHoundService::poll() should be in GlobalService.   It's time to merge libs.rs with randhound.rs. As a temporary solution I can suggest to merge this code back to  RandHoundService::poll().

Comment (rtsisyk) [reply randhound/src/randhound.rs]: Discussed and agreed to fix in a separate ticket.
https://github.com/stegos/stegos/issues/266

File: randhound/src/lib.rs [hunk 3]
```diff
@@ -297,19 +299,35 @@ impl Future for RandHoundService {
 
     fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
         loop {
+            let deadline_poll = self.state.poll_deadline();
```
Comment (vldm) [inline randhound/src/lib.rs]: If atleast one message was in `self.recv` when timeout expire, then we probably would ignore this deadline?

For example, imagine next flow:
1) `self.state.poll_deadline()` return's `Async::Ready(Some(...))` 
2) `self.recv.poll()` returns somethin other than `Async::NotReady`
After this next iteration of loop began and new call to `self.state.poll_deadline()` starts, old `deadline_poll` result will be forgetten.

Have i missing something?

Comment (echupriyanov) [reply randhound/src/lib.rs]: Result of deadline_poll() is processed in the match statement. On the next iteration we check if any more expirations happen

**Pull request merged**